### PR TITLE
update setup.py to use requirements.txt

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

﻿Update setup.py to use requirements.txt.
This is discussed in Terra Qiskit/qiskit-terra#1586.

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -1,3 +1,5 @@
+include LICENSE.txt
```
Comment (t-imamichi) [inline MANIFEST.in:1]: I don't think you need to manually include LICENSE.txt and requirements.txt because of `recursive-include ... *.txt`

Comment (ikkoham) [reply MANIFEST.in:1]: Thanks.

Comment (ikkoham) [reply MANIFEST.in:1]: I just found that recursive-include includes all *.txt files under qiskit_optimization. This line is necessary.

Comment (t-imamichi) [reply MANIFEST.in:1]: sorry. I misunderstood the spec. You are correct. https://packaging.python.org/guides/using-manifest-in/#manifest-in-commands

**Comment by manoelmarques** [comment]
I agree with this change, however the PR #45 adds the following line to requirements.txt:
```
--extra-index-url https://pypi.gurobi.com
```

This is one of those flags that only works on requirements.txt but will fail on setup.py. If that PR goes ahead with that then we cannot use requirements.txt in setup.py. I will add a hold to this PR pending a decision next meeting.

**Comment by t-imamichi** [comment]
I think gurobi should be extra_required. Then, it is not included requirements.txt and this PR works.

**Comment by manoelmarques** [comment]
Unfortunately because gurobipy is hosted on a private pypi site, it doesn’t work in setup.py via setuptools for security reasons. It can’t currently be in install_requires or extra_required. I will probably just remove it from requirements.txt in the other PR and the approve this one.

**Comment by t-imamichi** [comment]
I have a worry to include a private pypi site in requirements.txt because our CI may fail due to their server error.

**Comment by manoelmarques** [comment]
Exactly, that is why I want to remove it from requirements.txt in the other PR and let the users install themselves.

**Review by manoelmarques — APPROVED** [review]

**Pull request merged**